src/common/dom/apply_themes_on_element.ts

```diff
@@ -20,27 +21,34 @@ const hexToRgb = (hex: string): string | null => {
  *
  * element: Element to apply theme on.
  * themes: HASS Theme information
- * localTheme: selected theme.
- * updateMeta: boolean if we should update the theme-color meta element.
+ * selectedTheme: selected theme.
  */
 export const applyThemesOnElement = (
   element,
-  themes,
-  localTheme,
-  updateMeta = false
+  themes: HomeAssistant["themes"],
+  selectedTheme?: string
 ) => {
+  const newTheme = selectedTheme ? themes.themes[selectedTheme] : undefined;
+
+  // Styles that need to be reset from the previous theme
   if (!element._themes) {
+    if (!newTheme) {
+      // No styles to reset, and no styles to set
+      return;
+    }
     element._themes = {};
   }
-  let themeName = themes.default_theme;
-  if (localTheme === "default" || (localTheme && themes.themes[localTheme])) {
-    themeName = localTheme;
-  }
+
+  // Add previous set keys to reset them
   const styles = { ...element._themes };
-  if (themeName !== "default") {
-    const theme = { ...derivedStyles, ...themes.themes[themeName] };
+  if (newTheme) {
+    const theme = {
+      ...derivedStyles,
+      ...newTheme,
+    };
     Object.keys(theme).forEach((key) => {
       const prefixedKey = `--${key}`;
+      // Save key so we can reset it later if needed
       element._themes[prefixedKey] = "";
```

Comment:
We only need to start a new object. We don't have to keep all the properties of all the themes ever set here, because we will override them all in the previous applyTheme call.

Set Theme A which includes property 1, 2 sets property 1, 2
Set Theme B which includes property 2, 3 sets property 2, 3 and resets 1

**Currently:**
Set Theme C which includes property 4 sets property 4 and resets 1, 2, 3

**Proposed:**
Set Theme C which includes property 4 sets property 4 and resets 2, 3. Property 1 was already reset when theme B was applied.